modules/bash-commons/src/aws.sh

```diff
@@ -12,13 +12,39 @@ readonly metadata_endpoint="http://169.254.169.254/latest"
 readonly metadata_dynamic_endpoint="http://169.254.169.254/latest/dynamic"
 # The AWS EC2 Instance document endpoint
 readonly instance_identity_endpoint="http://169.254.169.254/latest/dynamic/instance-identity/document"
+# The AWS EC2 Instance IMDSv2 Token endpoint
+readonly imdsv2_token_endpoint="http://169.254.169.254/latest/api/token"
 # A convenience variable representing 3 hours, for use in requesting a token from the IMDSv2 endpoint
 readonly three_hours_in_s=10800
 # A convenience variable representing 6 hours, which is the maximum configurable session duration when requesting
 # a token from IMDSv2
 readonly six_hours_in_s=21600
-# By default, we use Instance Metadata service version 1. Although version 2 is preferred, version 1 is "fully secure" according to Amazon. We'll continue defaulting to version 1 as long as we're updating our dependent modules to take advantage of this new functionality in bash-commons. Once we've completed our migration, we will begin defaulting to version 2. Users can always specify the version of the Instance Metadata Service they want bash-commons to consult by setting the environment variable GRUNTWORK_BASH_COMMONS_IMDS_VERSION
-default_instance_metadata_version="1"
+
+# Detect if the instance is using IMDSv2 or if it is using IMDSv1 still.
+# Users can always specify the version of the Instance Metadata Service they want bash-commons
+# to consult by setting the environment variable GRUNTWORK_BASH_COMMONS_IMDS_VERSION
+# Defaults to IMDSv2 since that is now enabled by default on instances (IMDS only has two options,
+# "optional" = both v1 and v2, or "required" = v2 only).  All new instances support v2 now.
+curl -s -o /dev/null -X PUT ${imdsv2_token_endpoint} -H "X-aws-ec2-metadata-token-ttl-seconds: 10"
```

Comment:
This logic as its written fails if the curl command exits with a non-zero status. Can you wrap this output in a var so it won't stop the script from running if it encounters a non-zero exit code? 

For reference, you can test this with: `docker-compose up --exit-code-from bats_ubuntu2004_imds_v2 bats_ubuntu2004_imds_v2`

Right now this fails with: 

```
bash-commons-bats_ubuntu2004_imds_v2-1  | 1..1
bash-commons-bats_ubuntu2004_imds_v2-1  | not ok 1 bats-gather-tests
bash-commons-bats_ubuntu2004_imds_v2-1  | # (in file test/../modules/bash-commons/src/aws.sh, line 28,
bash-commons-bats_ubuntu2004_imds_v2-1  | #  in test file test/aws-cli.bats, line 3)
bash-commons-bats_ubuntu2004_imds_v2-1  | #   `source "$BATS_TEST_DIRNAME/../modules/bash-commons/src/aws.sh"' failed with status 7
bash-commons-bats_ubuntu2004_imds_v2-1 exited with code 1
Aborting on container exit...
```

Reply:
Ah, yes.  Ok.  That makes sense.
I think the Status 7 is from the endpoint being blocked in some way (firewal, disabled, etc).
I'll see what I can do.  Thanks for the hint on how to test it.